beamer/tests/util.py

```diff
@@ -163,20 +164,33 @@ def calculate_gas_spending():
     yield lambda: w3.eth.get_balance(address) + calculate_gas_spending() - balance_before
 
 
-def get_fee_data(request_manager):
-    return (
-        request_manager.protocolFeePPM(),
-        request_manager.lpFeePPM(),
-        request_manager.minLpFee(),
-    )
+# The function updates the token values partially whatever variable is in the params dictionary
+# This function works similar to adding a parameters dict to a transaction in web3.py
+def update_token(request_manager, token, params, *args, **kwargs):
+    token_data = list(request_manager.tokens(token.address))
+    for i, key in enumerate(["transfer_limit", "min_lp_fee", "lp_fee_ppm", "protocol_fee_ppm"]):
+        token_data[i] = params.get(key, token_data[i])
+    # the last element of token data is collectedProtocolFees which is not part of the update
+    request_manager.updateToken(token.address, *token_data[:-1], *args, **kwargs)
```

Comment:
Nitpick: I think we could avoid the `[:-1]` kludge by doing e.g.

```python
fields = "transfer_limit", "min_lp_fee", "lp_fee_ppm", "protocol_fee_ppm"
token_data = tuple(params.get(key, token_data[i]) for i, key in enumerate(fields))
request_manager.updateToken(token.address, *token_data, *args, **kwargs)
```

Reply:
nice!